### PR TITLE
feat: add progress bar for progress flag

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
No

**Summary**
Add a real progress bar for`--progress` flag.
image

**Does this PR introduce a breaking change?**
Yes

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/utils/Compiler.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const webpack = require('webpack');
 const logger = require('./logger');
+const readline = require('readline');
```
Comment (ematipico) [inline packages/webpack-cli/lib/utils/Compiler.js]: Looks like this is not used

File: packages/webpack-cli/lib/utils/Compiler.js [hunk 2]
```diff
@@ -15,17 +16,24 @@ class Compiler {
...
             if (outputOptions.progress) {
-                let tmpMsg;
+                process.stdout.write("\n");
                 const defaultProgressPluginHandler = (percent, msg) => {
                     percent = Math.floor(percent * 100);
-                    if (percent === 100) {
-                        msg = 'Compilation completed';
+                    process.stdout.clearLine();
+                    process.stdout.cursorTo(0);
+                    if (percent !== undefined) {
```
Comment (anshumanv) [inline packages/webpack-cli/lib/utils/Compiler.js]: Will `if (percent)` not work? Are there caveats to that?

File: packages/webpack-cli/lib/utils/Compiler.js [hunk 3]
```diff
@@ -15,17 +16,24 @@ class Compiler {
...
+                        process.stdout.write(' (')
+                        for(let i = 0; i <= 100; i+=10) {
```
Comment (anshumanv) [inline packages/webpack-cli/lib/utils/Compiler.js]:
```suggestion
                        for (let i = 0; i <= 100; i+=10) {
```

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/Compiler.js]: Is prettier not running for this file?

File: packages/webpack-cli/lib/utils/Compiler.js (hunk 2 above)
Comment (rishabh3112) [reply packages/webpack-cli/lib/utils/Compiler.js]: Zero as percent is a valid value, thus we should print when value is not strictly undefined.

File: packages/webpack-cli/lib/utils/Compiler.js (hunk 3 above)
Comment (rishabh3112) [reply packages/webpack-cli/lib/utils/Compiler.js]: Yup I think so, let me check.

File: packages/webpack-cli/lib/utils/Compiler.js (hunk 2 above)
Comment (anshumanv) [reply packages/webpack-cli/lib/utils/Compiler.js]: Ah cool, my bad. lgtm. 👍

File: packages/webpack-cli/lib/utils/Compiler.js (hunk 3 above)
Comment (ematipico) [reply packages/webpack-cli/lib/utils/Compiler.js]: Probably you did a commit and skipped the git hooks?

Comment (rishabh3112) [reply packages/webpack-cli/lib/utils/Compiler.js]: I had to run eslint with `--no-ignore` for it to lint.

**Pull request merged**